### ISSUE TITLE
Inserting multiple blocks in page has errors

### ISSUE THREAD

**Issue by SplicedWeb** [Samuell1/contenteditor-plugin#81]
Inserted
{% component 'contenteditor' file="filename_in_content1.htm" %}

Then later on page
{% component 'contenteditor' file="filename_in_content2.htm" fixture="h3" class="media" %}

Then again later on page 
{% component 'contenteditor' file="filename_in_content3.htm" tools="left-align,right-align" %}

the 3rd instance is rendered as h3 ?

**Comment by samuelpatro** [comment]
@SplicedWeb For temporary fix you can create other component and use only that for fixtures. I wasnt able to find a fix for this for now. Looks like components in octobercms are sharing properties together and that is the issue.

**Comment by SplicedWeb** [comment]
Thanks Samuell1, I'm not sure I understand your fix though.

**Comment by samuelpatro** [comment]
Like this. Just adding ContentEditor component in page and using this component name in your code.

![obrázok](https://user-images.githubusercontent.com/3110002/62005648-9fde3100-b136-11e9-8e46-3c5ea02d21f9.png)

**Comment by SplicedWeb** [comment]
Thanks so much for clarifying

**Comment by PubliAlex** [comment]
another solution is to declare a fixture="" to the third element, otherwise, it will get the fixture of the last used element. Same for class

{% component 'contenteditor' file="filename_in_content1.htm" %}

{% component 'contenteditor' file="filename_in_content2.htm" fixture="h3" class="media" %}

{% component 'contenteditor' file="filename_in_content3.htm" tools="left-align,right-align" fixture="" class="" %}

**Comment by samuelpatro** [comment]
@PubliAlex yep or that i was thinking to separate fixture and main editor to different components to prevent this issue but it will be backwards incompatible.

**Comment by jonathan-hc** [comment]
I just posted https://github.com/Samuell1/contenteditor-plugin/issues/101 and it looks like my problem is related. I have more than 10 instances of contentbuilder in my template in some cases, but I'm inserting the component in the layout only once. I do that because I don't want the entire HTML to be edited, only the text within individual **h?** or **p** tags Do I need to add the component 10 times in that case? That seems very impractical.

**Comment by samuelpatro** [comment]
@jonathan-hc Yes, its because one component in page is same component it doesnt initialize per tag, i tried to somehow to find a solution but still nothing.

**Comment by Conobi** [comment]
> I don't want the entire HTML to be edited, only the text within individual **h?** or **p** tags Do I need to add the component 10 times in that case? That seems very impractical.

Same issue here, it would be so cool to have a simpler syntax for editing a single tag, a bit like does this plugin: https://octobercms.com/plugin/fw-editme

**Comment by PubliAlex** [comment]
Another issue related to multiple blocks, it instanciate StylePalette multiple times :

![image](https://user-images.githubusercontent.com/55833027/175267658-0c389120-42e5-47ef-8756-205d5b4880d8.png)

As a result, styles are displayed multiple times :

![image](https://user-images.githubusercontent.com/55833027/175267802-fa86f16d-bdd1-4c6e-9d01-f5c4893e2729.png)

EDIT : I add a precision, it does that when the content editor is called inside partials, cause, for each partials, the renderCount variable restart to 1. That variable should be more "global" and not restart for the whole page

**Comment by PubliAlex** [comment]
Created a PR #117 that fix the issue of multiple StylePalette
